java-frontend/src/main/java/org/sonar/java/UCFGJavaVisitor.java

```diff
@@ -248,6 +254,21 @@ private void buildCall(Tree element, UCFGBuilder.BlockBuilder blockBuilder, Iden
     }
   }
 
+  private void buildConstructorInvocation(BlockBuilder blockBuilder, IdentifierGenerator idGenerator, NewClassTree newClassTree) {
+    List<Expression> arguments = null;
+    if(isString(newClassTree.constructorSymbol().owner().type()) || newClassTree.arguments().stream().map(ExpressionTree::symbolType).anyMatch(UCFGJavaVisitor::isString)) {
+      arguments = argumentIds(idGenerator, newClassTree.arguments());
+    }
+
+    if (arguments != null) {
```

Comment:
I think we can refactor these part in order to share the logic for method invocations and constructors call.